test/e2e/forking.t.js

```diff
@@ -0,0 +1,281 @@
+                trustedSequencer = await ethers.getSigner(trustedSequencerAddress);
+                await deployer.sendTransaction({
+                    to: trustedSequencerAddress,
+                    value: ethers.utils.parseEther('4'),
+                });
+                await expect(
+                    maticTokenContract.connect(trustedSequencer).approve(polygonZkEVMContract.address, maticAmount.mul(batchesNum)),
+                ).to.emit(maticTokenContract, 'Approval');
+                await maticTokenContract.transfer(trustedSequencer.address, ethers.utils.parseEther('100'));
+            }
+
+            // Sequence Batches
+            await expect(polygonZkEVMContract.connect(trustedSequencer).sequenceBatches([currentSequence], trustedSequencer.address))
+                .to.emit(polygonZkEVMContract, 'SequenceBatches')
+                .withArgs(Number(lastBatchSequenced) + 1);
+        }
+
+        // Set state and exit root
+        await polygonZkEVMContract.setStateRoot(inputJson.oldStateRoot, inputJson.oldNumBatch);
+
+        const { aggregatorAddress } = inputJson;
+        await ethers.provider.send('hardhat_impersonateAccount', [aggregatorAddress]);
+        const aggregator = await ethers.getSigner(aggregatorAddress);
+        await deployer.sendTransaction({
+            to: aggregatorAddress,
+            value: ethers.utils.parseEther('4'),
+        });
+        await polygonZkEVMContract.connect(admin).setTrustedAggregator(aggregatorAddress);
+
+        const batchAccInputHash = (await polygonZkEVMContract.sequencedBatches(inputJson.newNumBatch)).accInputHash;
+        expect(batchAccInputHash).to.be.equal(inputJson.newAccInputHash);
+
+        const proof = generateSolidityInputs(proofJson);
+
+        // Verify snark input
+        const circuitInputStarkJS = await calculateSnarkInput(
+            inputJson.oldStateRoot,
+            inputJson.newStateRoot,
+            inputJson.newLocalExitRoot,
+            inputJson.oldAccInputHash,
+            inputJson.newAccInputHash,
+            inputJson.oldNumBatch,
+            inputJson.newNumBatch,
+            inputJson.chainID,
+            inputJson.aggregatorAddress,
+            forkID,
+        );
+
+        expect(circuitInputStarkJS).to.be.eq(Scalar.e(input[0]));
+
+        // aggregator forge the batch
+        const { newLocalExitRoot } = inputJson;
+        const { newStateRoot } = inputJson;
+        const { oldNumBatch } = inputJson;
+        const { newNumBatch } = inputJson;
+        const pendingStateNum = 0;
+
+        // Verify batch
+        await expect(
+            polygonZkEVMContract.connect(aggregator).verifyBatchesTrustedAggregator(
+                pendingStateNum,
+                oldNumBatch,
+                newNumBatch,
+                newLocalExitRoot,
+                newStateRoot,
+                Array.from(Array(24)).fill('0x0e7073c1e73dfb716c35623b741e4ccfc6290d943f3df60377ad8799373ae439'),
```

Comment:
Is this just an arbitrary hash to test that it fails?

Reply:
yes